### DISCUSSION TITLE
Extracting the each highlighted pages and gather all of them in one pdf

### DISCUSSION THREAD

**Discussion by z8002** [py-pdf/pypdf#2584] [Show and tell]
## Explanation

i want to extract  highligted pages from a pdf and save the extracted pages in one piece as a output.pdf .  is it possible to add this feature  as a function ? 
 
## Code Example

```
def extract_highlighted_pages(input_pdf_path, output_pdf_path):
    highlighted_pages = set()

    # Open the PDF file in binary mode
    pdf_file = open(input_pdf_path, 'rb')

    pdf_reader = pypdf.PdfReader(pdf_file)
    # Iterate through each page
    for page_num in range(len(pdf_reader.pages)):
        page = pdf_reader.pages[page_num]

        # Check if the page has annotations/highlights
        if '/Annots' in page:
            annots = page['/Annots']
            for annot_ref in annots:
                annot = pdf_reader.get_object(annot_ref)
                if annot.get('/Subtype') == '/Highlight':
                    highlighted_pages.add(page_num + 1)  # Page numbers start from 1


    # Create a new PDF with only the highlighted pages
    output_pdf =  open(output_pdf_path, 'wb')
    pdf_writer = pypdf.PdfWriter()

    for page_numf in highlighted_pages:
        pdf_writer.add_page(pdf_reader.pages[page_numf-1])  # Adjust page number

    pdf_writer.write(output_pdf)
```

```python
from pypdf import PdfReader, PdfWriter

**Comment by z8002** [comment]
if you open a pr that would be super good for me

**Comment by stefan6419846** [comment]
This does not seem like something which should go into the core library, but rather into https://github.com/py-pdf/pdfly

**Comment by pubpub-zz** [comment]
@z8002 
highlighting is actually an area colored with semi-transparent color. it is not link with the text beneath. The solution is to use a visitor function in the extract text (https://pypdf.readthedocs.io/en/stable/user/extract-text.html#using-a-visitor) in order to identify whether the text is within the area of the highlight. some lines partially highlighted may be tricky to cut

**Comment by stefan6419846** [comment]
I tend to close this issue or convert it to a discussion as this is not really something which should go into *pypdf* directly.

**Comment by pubpub-zz** [comment]
as proposed by @stefan6419846, I convert it into discussion
